packages/client/hooks/useLinearProjectsAndTeams.ts

```diff
@@ -0,0 +1,154 @@
+import graphql from 'babel-plugin-relay/macro'
+import {useMemo, useState} from 'react'
+import {useFragment} from 'react-relay'
+import {
+  useLinearProjectsAndTeams_teamMember$data,
+  useLinearProjectsAndTeams_teamMember$key
+} from '../__generated__/useLinearProjectsAndTeams_teamMember.graphql'
+import getNonNullEdges from '../utils/getNonNullEdges'
+import getUniqueEdges from '../utils/getUniqueEdges'
+import {isNotNull} from '../utils/predicates'
+
+type ArrayElement<ArrayType extends readonly unknown[] | null | undefined> =
+  ArrayType extends readonly (infer ElementType)[] ? ElementType : never
+
+type LinearQuery = useLinearProjectsAndTeams_teamMember$data extends {
+  integrations?: {linear?: {api?: {query?: infer Q | null} | null} | null} | null
+}
+  ? Q
+  : never
+
+type AllProjectEdges = LinearQuery extends {allProjects?: {edges?: infer E | null} | null}
+  ? E
+  : never
+
+type TeamEdges = LinearQuery extends {teams?: {edges?: infer E | null} | null} ? E : never
+
+type ProjectEdge = NonNullable<ArrayElement<AllProjectEdges>>
+type TeamEdge = NonNullable<ArrayElement<TeamEdges>>
+
+export type Project = NonNullable<ProjectEdge['node']>
+export type Team = NonNullable<TeamEdge['node']>
+
+export type LinearProjectOrTeam = Project | Team
+
+const getNodeId = (edge: ProjectEdge | TeamEdge) => {
+  return edge?.node?.id
+}
+
+const linearProjectNameWithTeam = (project: Project): string => {
+  const {name: projectName, teams} = project
+  const teamName = teams?.nodes?.[0]?.displayName
+  return teamName ? `${teamName}/${projectName}` : projectName || 'Unknown Project'
+}
+
+const getItemName = (item: LinearProjectOrTeam): string => {
```

Comment:
+1 I'm surprised to find this here when it's also redefined in consumers of `useLinearProjectsAndTeams`